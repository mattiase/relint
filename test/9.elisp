;;; Relint test file 9          -*- emacs-lisp -*-

;; Test mutation/binding of known variables for detecting regexps.

(defun test-9 ()
  (setq-local page-delimiter "[aa]")
  (setq-local paragraph-separate "[bb]")
  (setq-local paragraph-start "[cc]")
  (setq-local sentence-end "[dd]")
  (setq-local comment-start-skip "[ee]")
  (setq-local comment-end-skip "[ff]")

  (setq sentence-end "[gg]")
  (set (make-local-variable 'paragraph-start) "[hh]")

  (let ((paragraph-separate "[ii]")
        (page-delimiter "[jj]"))
    (let* ((comment-start-skip "[kk]")
           (comment-end-skip "[ll]"))
      (asdf)))

  (setq treesit-sentence-type-regexp "+0")
  (setq treesit-sexp-type-regexp "+1"))

(defvar bad-imenu
  '(("*less*" "" 0)
    (nil "+b+" 0)))

(defvar bad-treesit-imenu
  '((nil "node" nil nil)
    ("+++" "+c+" nil nil)))

(defun test-9-ge ()
  (setq-local imenu-generic-expression
              '((nil "oh" 0)
                ("*more*" "+a+" 0)))
  (setq imenu-generic-expression bad-imenu)
  (setq-local treesit-simple-imenu-settings bad-treesit-imenu)
  (set (make-local-variable 'treesit-simple-imenu-settings)
       '((nil "type" nil nil)
         ("+more+" "+d+" nil nil))))

(defun test-9-fl-kw ()
  (setq-local font-lock-keywords '(("[mm]" . tag)))
  (setq font-lock-keywords '(("[nn]" . tag)))
  (set (make-local-variable 'font-lock-keywords) '(("[oo]" . tag))))

(defconst my-fl-keyw-1
  '(("[pp]" . alpha)))

(defconst my-font-lock-keywords-2
  '(("[qq]" . beta)))

(defun test-9-fl-def ()
  (setq font-lock-defaults
        '((my-fl-keyw-1
           my-font-lock-keywords-2)
          moo mooo)))

(defun test-9-ts-indent ()
  (setq treesit-simple-indent-rules
        '((a ((match "+e+" "+f+" "+g+") a 0)
             ((n-p-gp "+h+" "+i+" "+j+") a 0))
          (b ((parent-is "+k+") b 0)
             ((node-is "+l+") b 0)
             ((field-is "+m+") b 0)))))

(defun test-9-ts-defun ()
  (setq treesit-defun-type-regexp "+n+")
  (setq treesit-defun-type-regexp (cons "+o+" #'ignore)))
