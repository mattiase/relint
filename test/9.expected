9.elisp:6:33: In page-delimiter: Duplicated `a' inside character alternative (pos 2)
  "[aa]"
   ..^
9.elisp:7:37: In paragraph-separate: Duplicated `b' inside character alternative (pos 2)
  "[bb]"
   ..^
9.elisp:8:34: In paragraph-start: Duplicated `c' inside character alternative (pos 2)
  "[cc]"
   ..^
9.elisp:9:31: In sentence-end: Duplicated `d' inside character alternative (pos 2)
  "[dd]"
   ..^
9.elisp:10:37: In comment-start-skip: Duplicated `e' inside character alternative (pos 2)
  "[ee]"
   ..^
9.elisp:11:35: In comment-end-skip: Duplicated `f' inside character alternative (pos 2)
  "[ff]"
   ..^
9.elisp:13:25: In sentence-end: Duplicated `g' inside character alternative (pos 2)
  "[gg]"
   ..^
9.elisp:14:50: In paragraph-start: Duplicated `h' inside character alternative (pos 2)
  "[hh]"
   ..^
9.elisp:16:32: In paragraph-separate: Duplicated `i' inside character alternative (pos 2)
  "[ii]"
   ..^
9.elisp:17:28: In page-delimiter: Duplicated `j' inside character alternative (pos 2)
  "[jj]"
   ..^
9.elisp:18:35: In comment-start-skip: Duplicated `k' inside character alternative (pos 2)
  "[kk]"
   ..^
9.elisp:19:33: In comment-end-skip: Duplicated `l' inside character alternative (pos 2)
  "[ll]"
   ..^
9.elisp:22:39: In treesit-sentence-type-regexp: Unescaped literal `+' (pos 0)
  "+0"
   ^
9.elisp:23:35: In treesit-sexp-type-regexp: Unescaped literal `+' (pos 0)
  "+1"
   ^
9.elisp:36:28: In imenu-generic-expression: Unescaped literal `+' (pos 0)
  "+a+"
   ^
9.elisp:37:34: In imenu-generic-expression: Unescaped literal `+' (pos 0)
  "+b+"
   ^
9.elisp:38:45: In treesit-simple-imenu-settings: Unescaped literal `+' (pos 0)
  "+c+"
   ^
9.elisp:41:21: In treesit-simple-imenu-settings: Unescaped literal `+' (pos 0)
  "+d+"
   ^
9.elisp:44:40: In font-lock-keywords (tag): Duplicated `m' inside character alternative (pos 2)
  "[mm]"
   ..^
9.elisp:45:34: In font-lock-keywords (tag): Duplicated `n' inside character alternative (pos 2)
  "[nn]"
   ..^
9.elisp:46:56: In font-lock-keywords (tag): Duplicated `o' inside character alternative (pos 2)
  "[oo]"
   ..^
9.elisp:52:9: In my-font-lock-keywords-2 (beta): Duplicated `q' inside character alternative (pos 2)
  "[qq]"
   ..^
9.elisp:56:9: In font-lock-defaults (alpha): Duplicated `p' inside character alternative (pos 2)
  "[pp]"
   ..^
9.elisp:62:23: In treesit-simple-indent-rules (a): Unescaped literal `+' (pos 0)
  "+e+"
   ^
9.elisp:62:29: In treesit-simple-indent-rules (a): Unescaped literal `+' (pos 0)
  "+f+"
   ^
9.elisp:62:35: In treesit-simple-indent-rules (a): Unescaped literal `+' (pos 0)
  "+g+"
   ^
9.elisp:63:24: In treesit-simple-indent-rules (a): Unescaped literal `+' (pos 0)
  "+h+"
   ^
9.elisp:63:30: In treesit-simple-indent-rules (a): Unescaped literal `+' (pos 0)
  "+i+"
   ^
9.elisp:63:36: In treesit-simple-indent-rules (a): Unescaped literal `+' (pos 0)
  "+j+"
   ^
9.elisp:64:27: In treesit-simple-indent-rules (b): Unescaped literal `+' (pos 0)
  "+k+"
   ^
9.elisp:65:25: In treesit-simple-indent-rules (b): Unescaped literal `+' (pos 0)
  "+l+"
   ^
9.elisp:66:26: In treesit-simple-indent-rules (b): Unescaped literal `+' (pos 0)
  "+m+"
   ^
9.elisp:69:36: In treesit-defun-type-regexp: Unescaped literal `+' (pos 0)
  "+n+"
   ^
9.elisp:70:35: In treesit-defun-type-regexp: Unescaped literal `+' (pos 0)
  "+o+"
   ^
